docs/core/extensions/file-globbing.md

```diff
@@ -0,0 +1,248 @@
+```
+
+The preceding C# code:
+
+- Instantiates a <xref:Microsoft.Extensions.FileSystemGlobbing.Matcher> object.
+- Calls <xref:Microsoft.Extensions.FileSystemGlobbing.MatcherExtensions.AddIncludePatterns(Microsoft.Extensions.FileSystemGlobbing.Matcher,System.Collections.Generic.IEnumerable{System.String}[])> to add several file name patterns to include.
+- Declares and assigns the search directory value.
+- Calls `GetResultsInFullPath` given the `searchDirectory` value to yield all matching files as a `IEnumerable<string>`.
+
+### Match overloads
+
+The <xref:Microsoft.Extensions.FileSystemGlobbing.PatternMatchingResult> object represents a collection of <xref:Microsoft.Extensions.FileSystemGlobbing.FilePatternMatch> instances, and exposes a `boolean` value indicating whether the result has matches &mdash; <xref:Microsoft.Extensions.FileSystemGlobbing.PatternMatchingResult.HasMatches?displayProperty=nameWithType>.
+
+With a `Matcher` instance, you can call the any of the various `Match` overloads to get a pattern matching result. The `Match` methods invert the responsibility on the caller to provide a file or a collection of files in which to evaluate for matches. In other words, the caller is responsible for passing the file to match on.
+
+> [!IMPORTANT]
+> When using any of the `Match` overloads, there is no file system I/O involved. All of the file globbing is done in memory with the include and exclude patterns of the `matcher` instance. The parameters of the `Match` overloads do not have to be fully qualified paths. The current directory (<xref:System.IO.Directory.GetCurrentDirectory?displayProperty=nameWithType>) is used when not specified.
+
+To match a single file:
+
+```csharp
+Matcher matcher = new();
+matcher.AddInclude("**/*.md");
+
+PatternMatchingResult result = matcher.Match("file.md");
+```
+
+The preceding C# code:
+
+- Matches any file with the *.md* file extension, at an arbitrary directory depth.
+- If a file named *file.md* exists in a subdirectory from the current directory:
+  - `result.HasMatches` would be `true`.
+  - and `result.Files` would have one match.
+
+The additional `Match` overloads work in similar ways.
+
+## Pattern formats
+
+The patterns that are specified in the `AddExclude` and `AddInclude` methods can use the following formats to match multiple files or directories.
+
+- Exact directory or file name
+  
+  - `some-file.txt`
+  - `path/to/file.txt`
+
+- Wildcards `*` in file and directory names that represent zero to many characters not including separator characters.
+
+    | Value          | Description                                                            |
+    |----------------|------------------------------------------------------------------------|
+    | `*.txt`        | All files with *.txt* file extension.                                  |
+    | `*.*`          | All files with an extension.                                           |
```

Comment:
"with an extension" seems to imply that \* means 1-to-many, rather than 0-to-many as stated above.

Reply:
How would you re-phrase this?

Reply:
Disregard -- I was thinking a file name could end in dot with nothing following, but on trying to create one like that I find that it just discards the dot.